### PR TITLE
nicotine-plus: 3.3.5 -> 3.3.6

### PR DESCRIPTION
**Pull request by amadaluzia**
I have done some refactoring in the nicotine-plus package to fix libadwaita being an issue, (you may test this with running nicotine-plus with NICOTINE_LIBADWAITA="1") and updated the package from 3.3.5 to 3.3.6.

You may read about changes made in 3.3.6 here: https://nicotine-plus.org/NEWS.html

This introduces a renderer change which may affect some computers.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by amadaluzia** [comment]
Damnit, emacs...

Will try and editorconfig the package, apologies.

File: pkgs/by-name/ni/nicotine-plus/package.nix [hunk 1]
```diff
@@ -1,42 +1,51 @@
...
+    glib
+    gdk-pixbuf
     gtk4
+  ];
+
+  buildInputs = [
+    libadwaita
+  ];
+
+  propagatedBuildInputs = [
```
Comment (pluiedev) [inline pkgs/by-name/ni/nicotine-plus/package.nix:37]:
```suggestion
  dependencies = [
```

File: pkgs/by-name/ni/nicotine-plus/package.nix [hunk 2]
```diff
@@ -1,42 +1,51 @@
...
-  version = "3.3.5";
+  version = "3.3.6";
 
   src = fetchFromGitHub {
     owner = "nicotine-plus";
     repo = "nicotine-plus";
     rev = "refs/tags/${version}";
-    hash = "sha256-6tA3d+QX2ArDH4aeWZNKuIXe3Sk32JaFe8d0C8G9Akc=";
+    hash = "sha256-je3hyxbF9wKW2gvHoDp712EJxBxooS2z0pQM57WDdOk=";
   };
```
Comment (pluiedev) [inline pkgs/by-name/ni/nicotine-plus/package.nix:23]:
```suggestion
  build-system = [ python3Packages.setuptools ];
```

**Review by pluiedev — APPROVED** [review]
LGTM & Welcome to Nixpkgs! While you're here, why not (kindly) add some things that would make it even better?

File: pkgs/by-name/ni/nicotine-plus/package.nix [hunk 3]
```diff
@@ -1,42 +1,51 @@
...
+  gtk4,
+  glib,
+  python3Packages,
+  libadwaita,
 }:
-
 python3Packages.buildPythonApplication rec {
   pname = "nicotine-plus";
-  version = "3.3.5";
+  version = "3.3.6";
```
[the remark this answers is not in the document]
Comment (amadaluzia) [reply pkgs/by-name/ni/nicotine-plus/package.nix:15]: Added this.

File: pkgs/by-name/ni/nicotine-plus/package.nix (hunk 2 above)
Comment (amadaluzia) [reply pkgs/by-name/ni/nicotine-plus/package.nix:23]: Added this.

**Comment by amadaluzia** [comment]
Squashed all the commits together, I believe this is ready for merging now.

File: pkgs/by-name/ni/nicotine-plus/package.nix (hunk 1 above)
Comment (amadaluzia) [reply pkgs/by-name/ni/nicotine-plus/package.nix]: Added this.

**Review by pluiedev — APPROVED** [review]

**Pull request merged**